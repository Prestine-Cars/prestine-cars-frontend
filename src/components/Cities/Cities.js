// @ts-nocheck
import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { getCities } from '../../redux/reducers/cities';
import classes from '../modules/Cities.module.css';

const Cities = () => {
  const cities = useSelector((state) => state.cities);
  const { loading, error, cities: citiesData } = cities;
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCities());
  }, []);

  return (
    <>
      <div className="text-center m-4 p-4">
        <h2 className="text-center m-2 p-2">These are the available Cities</h2>
        <h3 className="text-center mt-2">
          Choose a country flag to see available cars
        </h3>
      </div>
      {loading && (
        <div className="d-flex flex-column justify-content-center align-items-center vh-100">
          <span className={`spinner-border ${classes.green}`} />
          <p>Loading the cities...</p>
        </div>
      )}
      {' '}
      {error && (
        <div className="alert alert-danger" role="alert">
          {error}
        </div>
      )}
      {' '}
      <div className="container">
        <div className="row">
          {citiesData.map((city) => (
            <div
              key={city.id}
              className={`col-md-6 m-2 d-flex justify-content-center align-items-center flex-column ${classes.img_wrap}`}
            >
              <Link to={`/cities/${city.id}`}>
                <img
                  className={`shadow p-3 mb-2 bg-body rounded ${classes.img}`}
                  alt={city.name}
                  src={city.flag_icon}
                />
              </Link>
              <hr className="mx-auto border-t-4 border-dotted w-32" />
              <h5 className="my-2 h5">{city.name}</h5>
            </div>
          ))}
        </div>
      </div>
    </>
  );
};

export default Cities;
