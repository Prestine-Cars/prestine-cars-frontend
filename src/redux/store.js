import { configureStore } from '@reduxjs/toolkit';
import userReducer from './reducers/user';
import cityReducer from './reducers/city';
import citiesReducer from './reducers/cities';
import carReducer from './reducers/car';

const store = configureStore({
  reducer: {
    user: userReducer,
    cities: citiesReducer,
    city: cityReducer,
    car: carReducer,
  },
});

export default store;
