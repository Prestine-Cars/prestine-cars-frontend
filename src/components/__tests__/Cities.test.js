import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Cities from '../Cities/Cities';
import store from '../../redux/store';

describe('Test presence of Cities', () => {
  test('should render Cities component', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Cities />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
