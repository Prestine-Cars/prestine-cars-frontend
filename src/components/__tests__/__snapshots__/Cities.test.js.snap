// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test presence of Cities should render Cities component 1`] = `
Array [
  <div
    className="cities_header"
  >
    <h3
      className="m-2 p-2"
    >
      Our services are in the following cities
    </h3>
    <h4
      className="mt-4"
    >
      Choose a country flag to see available cars
    </h4>
  </div>,
  <div
    className="d-flex flex-column justify-content-center align-items-center vh-100"
  >
    <span
      className="spinner-border green"
    />
    <p>
      Loading the cities...
    </p>
  </div>,
  " ",
  " ",
  <div
    className="container"
  >
    <div
      className="row"
    />
  </div>,
]
`;
